server/src/main/java/org/opensearch/transport/StreamTransportService.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.transport;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.common.Nullable;
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.common.transport.BoundTransportAddress;
+import org.opensearch.core.transport.TransportResponse;
+import org.opensearch.tasks.Task;
+import org.opensearch.telemetry.tracing.Tracer;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.util.function.Function;
+
+import static org.opensearch.discovery.HandshakingTransportAddressConnector.PROBE_CONNECT_TIMEOUT_SETTING;
+import static org.opensearch.discovery.HandshakingTransportAddressConnector.PROBE_HANDSHAKE_TIMEOUT_SETTING;
+
+/**
+ * Transport service for streaming requests, handling StreamTransportResponse.
+ * @opensearch.internal
+ */
+public class StreamTransportService extends TransportService {
+    private static final Logger logger = LogManager.getLogger(StreamTransportService.class);
+    public static final Setting<TimeValue> STREAM_TRANSPORT_REQ_TIMEOUT_SETTING = Setting.timeSetting(
+        "transport.stream.request_timeout",
+        TimeValue.timeValueMinutes(5),
+        Setting.Property.NodeScope,
+        Setting.Property.Dynamic
+    );
+
+    private volatile TimeValue streamTransportReqTimeout;
+
+    public StreamTransportService(
+        Settings settings,
+        Transport streamTransport,
+        ThreadPool threadPool,
+        TransportInterceptor transportInterceptor,
+        Function<BoundTransportAddress, DiscoveryNode> localNodeFactory,
+        @Nullable ClusterSettings clusterSettings,
+        TransportService transportService,
```

Comment:
Ok, this is weird. StreamTransportService extends TransportService but also takes a TransportService as a parameter?

Reply:
Yes, will remove it and pass individual objects. 
Ideally, I would like to refactor the TransportService itself to make it an interface and add the right APIs which should clean up and make it more manageable. But thats a bigger refactor and need to give some more thoughts.